cEP-0015.md

```diff
@@ -0,0 +1,98 @@
+# corobo Enhancement (security, tests and configurability)
+
+| Metadata |                                                          |
+| -------- | -------------------------------------------------------- |
+| cEP      | 15                                                       |
+| Version  | 1.0                                                      |
+| Title    | corobo Enhancement (security, tests and configurability) |
+| Authors  | Nitanshu Vashistha <mailto:nitanshu.vzard@gmail.com>     |
+| Status   | Proposed                                                 |
+| Type     | Process                                                  |
+
+## Abstract
+
+This cEP describes the details of enhancement of
+[corobo](https://github.com/coala/corobo) in terms of security, tests,
+configurability and the new plugins that are to be added to corobo as part
+of the
+[GSoC Project](https://summerofcode.withgoogle.com/projects/#6603667076022272).
+
+## Security
+
+### Why improve security?
+
+Security has been one of the major concerns due to some past experiences.
+We want to provide access controls based on user’s GitHub team memberships.
+
+### How to improve security?
+
+1. Make all LabHub commands except invite require being a member of the
+   organization.
+
+2. Add ability to ban from all gitter rooms at once.
+
+3. Remove auto-invite and give invite permissions to developers.
+
+4. Prevent newcomers from self-assigning themselves and give developers
+   permission to assign newcomers.
+
+5. Prevent corobo being used to spam a room.
+
+6. Force newcomers to finish one issue first.
+
+7. Require newcomers to find a newcomer issue to work on before they are
+   invited to join the organization.
+
```

Comment:
maybe even ban newcomers for a day or two, if they violate too many rules given in the newcomer's guide

Reply:
Won't that be too harsh :P ?
The developers/maintainers will invite them and assign the issue only after they find one to work on.

Reply:
thats why just for a day or two, asking them nicely to again go through the newcomer's guide and some git tutorials and such

Reply:
anyways its your call

Reply:
if they are doing so many things wrong, means they have taken this lightly and haven't gone through the provided content. also asking too many googlable questions may count to that

Reply:
I agree with your point but that might scare them away from contributing :P 
We can definitely add a command to warn them about not following the guidelines.
I think we'll need some discussion on this point.

Reply:
as I said its completely your call on that 👍

Reply:
We can add a scarscatic explaination for `corobo explain google` :P